### PR TITLE
feat: added tooltip and scale-icon in data-grid button

### PR DESCRIPTION
**Pull request by sameetah**
This feauture implements extension for the data-grid action cells to add tooltip and icons to the configured rows. This addresses the [Issue 2236](https://github.com/telekom/scale/issues/2236) that requested such feature. The tooltip  and icons use the existing icon and tooltip feature from the scale button. 
Reviews are appreciated.

### CODE REVIEW

**Review by maomaoZH — APPROVED** [review]

**Comment by maomaoZH** [comment]
Thanks for the PR! There are some build errors. Would you please have a look and fix them?

**Comment by sameetah** [comment]
@maomaoZH I have pushed the committed changes. There were some checks executed first time i created the pull request e.g. prettier, visual tests etc. Now I made some changes and pushed new commits to this pull request. Could you please let me know how these checks can be executed again in order to say now everything is ok?

**Comment by sameetah** [comment]
@acstll @felix-ico Could you please approve for the pipeline to run or do I need to do anything in order for pipelines to run?

File: packages/components/tsconfig.json [hunk 1]
```diff
@@ -11,7 +11,8 @@
     "noUnusedLocals": true,
     "noUnusedParameters": true,
     "jsx": "react",
-    "jsxFactory": "h"
+    "jsxFactory": "h",
+    "skipLibCheck": true
```
Comment (felix-ico) [inline packages/components/tsconfig.json:15]: is this required?

**Comment by felix-ico** [comment]
HI @sameetah looks like `build-pr / prettier ` and `build-pr / uncommitted-changes` are still failing. To fix the first one you should run `yarn format` and commit the changes, for the second one `yarn build` and commit the changes. `build-pr / visual-tests (pull_request)` should be OK to ignore, if there have been no visual changes in the components

**Comment by sameetah** [comment]
@acstll @felix-ico I have addressed the issues mentioned. 
For the 'build-pr/prettier' error, I ran 'yarn format' locally and committed the changes. 
For the 'build-pr/uncommitted-changes' error, I ran 'yarn build' locally and committed the changes. 
The changes have been pushed to the branch. Could you please review again? Thank you!

**Comment by felix-ico** [comment]
@sameetah the uncommited-changes job is still failing, could you make sure that all files have been commmited? 
Screenshot 2024-04-03 at 14 05 09

also, could you check if it's possible to remove the `skipLibCheck` you added?

**Comment by sameetah** [comment]
@felix-ico I removed the skipLibCheck rule in tsconfig file. Could you please have a look and see if checks pass now?

**Comment by sameetah** [comment]
@felix-ico The uncommitted changes are a single read me that is auto generated during the build process. All my changes are committed when I push so how do I account for the build script?

**Comment by sameetah** [comment]
@felix-ico Could you please now check if the uncommitted changes check is passing? I made some changes and this should hopefully work

**Comment by felix-ico** [comment]
@sameetah now you need to run `yarn format` and `yarn build` in `packages/components` again...

**Comment by sameetah** [comment]
@felix-ico Could you please check again if the checks pass this time?

**Comment by amir-ba** [comment]
closed in favor of https://github.com/telekom/scale/pull/2308

**Pull request closed without merging**